receiver/prometheusreceiver/config.go

```diff
@@ -47,103 +35,61 @@ type Config struct {
 	// ReportExtraScrapeMetrics - enables reporting of additional metrics for Prometheus client like scrape_body_size_bytes
 	ReportExtraScrapeMetrics bool `mapstructure:"report_extra_scrape_metrics"`
 
-	TargetAllocator *targetAllocator `mapstructure:"target_allocator"`
-
-	// ConfigPlaceholder is just an entry to make the configuration pass a check
-	// that requires that all keys present in the config actually exist on the
-	// structure, ie.: it will error if an unknown key is present.
-	ConfigPlaceholder any `mapstructure:"config"`
+	TargetAllocator *TargetAllocator `mapstructure:"target_allocator"`
 
 	// EnableProtobufNegotiation allows the collector to set the scraper option for
 	// protobuf negotiation when conferring with a prometheus client.
 	EnableProtobufNegotiation bool `mapstructure:"enable_protobuf_negotiation"`
 }
 
-type targetAllocator struct {
-	Endpoint    string        `mapstructure:"endpoint"`
-	Interval    time.Duration `mapstructure:"interval"`
-	CollectorID string        `mapstructure:"collector_id"`
-	// ConfigPlaceholder is just an entry to make the configuration pass a check
-	// that requires that all keys present in the config actually exist on the
-	// structure, ie.: it will error if an unknown key is present.
-	ConfigPlaceholder any                `mapstructure:"http_sd_config"`
-	HTTPSDConfig      *promHTTP.SDConfig `mapstructure:"-"`
+// Validate checks the receiver configuration is valid.
+func (cfg *Config) Validate() error {
+	if (cfg.PrometheusConfig != nil || len(cfg.PrometheusConfig.ScrapeConfigs) == 0) && cfg.TargetAllocator == nil {
```

Comment:
```suggestion
	if (cfg.PrometheusConfig == nil || len(cfg.PrometheusConfig.ScrapeConfigs) == 0) && cfg.TargetAllocator == nil {
```

This still needs to be fixed here.

Reply:
This looks indeed odd still. There are unit tests failing because of this:

```
receivers::prometheus: no Prometheus scrape_configs or target_allocator set
```

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7478660342/job/20354137841?pr=29901

Reply:
I think was a glitch somewhere else. Not sure why failed that way.